test/metabase/pulse_test.clj

```diff
@@ -145,69 +147,51 @@
   [common & {:as message->m}]
   (doseq [[message m] message->m]
     (testing message
-      (do-test (merge-with merge common m)))))
+      (do-test! (merge-with merge common m)))))
 
 (def ^:private test-card-result {pulse.test-util/card-name true})
-(def ^:private test-card-regex  (re-pattern pulse.test-util/card-name))
+(def ^:private test-card-regex (re-pattern pulse.test-util/card-name))
 
 
 (defn- produces-bytes? [{:keys [rendered-info]}]
   (when rendered-info
     (pos? (alength (or (render/png-from-render-info rendered-info 500)
                        (byte-array 0))))))
 
-(defn- email-body? [{message-type :type, ^String content :content}]
-  (and (= "text/html; charset=utf-8" message-type)
-       (string? content)
-       (.startsWith content "<!doctype html>")))
-
-(defn- attachment? [{message-type :type, content-type :content-type, content :content}]
-  (and (= :inline message-type)
-       (= "image/png" content-type)
-       (instance? java.net.URL content)))
-
-(defn- add-rasta-attachment
-  "Append `attachment` to the first email found for Rasta"
-  [email attachment]
-  (update-in email ["rasta@metabase.com" 0] #(update % :body conj attachment)))
-
-
 ;;; +----------------------------------------------------------------------------------------------------------------+
 ;;; |                                                     Tests                                                      |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
 (deftest basic-timeseries-test
-  (do-test
+  (do-test!
    {:card
     (merge
-     (pulse.test-util/checkins-query-card {:breakout [!day.date]})
+     (pulse.test-util/checkins-query-card {:breakout [!day.date]
+                                           :limit 1})
      {:visualization_settings {:graph.dimensions ["DATE"]
                                :graph.metrics    ["count"]}})
-    :pulse   {:skip_if_empty false}
-
     :assert
     {:email
-     (fn [_ _]
-       (is (= (rasta-pulse-email)
-              (mt/summarize-multipart-email #"Pulse Name"))))
+     (fn [_ [email]]
```

Comment:
the result here is a list of email, since this test only send one we can take the first.